### PR TITLE
[LE-709] Added index argument to check_function and wrote the tests

### PR DESCRIPTION
**Pull request by BogdanFloris**
- index argument added to check_function
- also added the index argument to has_code, passed from check_function
- the index argument is used in the if-else statement in the match function.
- tests can be found test_check_function

### CODE REVIEW

File: tests/test_check_function.py [hunk 1]
```diff
@@ -23,7 +23,41 @@ def user_data_normalize():
...
+        ("B2", "MEDIAN", 0, True),   # 1
+        ("B2", "MEDIAN", 1, True),   # 1
+        ("B2", "MEDIAN", 2, False),  # 0
+        ("B2", "TEST", 0, False),    # 0
+        ("B2", "TEST", 1, False),    # 0
+        ("B2", "TEST", 2, False),    # 0
+        ("B2", "SUM", 0, False),     # 0
+    ]
+)
+def test_check_function_index(user_data_index, sct_range, function, index, correct):
```
Comment (hermansje) [inline tests/test_check_function.py]: This should be moved to be below `# Tests` (above it should be fixtures).

File: sheetwhat/checks/check_funcs.py [hunk 2]
```diff
@@ -24,12 +24,15 @@ def check_range(state, field, field_msg, missing_msg=None):
     )
 
 
-def has_code(state, pattern, fixed=False, incorrect_msg=None, normalize=lambda x: x):
+def has_code(state, pattern, fixed=False, index=0, incorrect_msg=None, normalize=lambda x: x):
```
Comment (hermansje) [inline sheetwhat/checks/check_funcs.py]: Adding an index argument to `has_code` is an interesting idea. We should create a ticket to implement it in other xwhat packages as well then.

File: sheetwhat/checks/check_funcs.py [hunk 3]
```diff
@@ -24,12 +24,15 @@ def check_range(state, field, field_msg, missing_msg=None):
...
     child = check_range(state, field="formulas", field_msg="formula")
 
     def match(on_text):
         if fixed:
-            return normalize(pattern) in str(on_text)
+            if index == 0:
```
Comment (hermansje) [inline sheetwhat/checks/check_funcs.py]: Is it necessary to handle `index=0` separately?

File: tests/test_check_function.py [hunk 4]
```diff
@@ -23,7 +23,41 @@ def user_data_normalize():
...
+        check_function(s, name=function, index=index)
+
+
+@pytest.fixture()
+def user_data_index():
+    return {
+        "values": [[1, 1, 1], [1, 52, 8]],
+        "formulas": [
+            ["=SUM(A1) + SUM(A1)", 1, 1],
+            ["AVERAGE($C$2) + AVERAGE($C$2)", "MEDIAN(B2:C5) + MEDIAN(B2:C5)"]
```
Comment (hermansje) [inline tests/test_check_function.py]: Testing mixed casing for the same function in one formula could be a good test.

File: sheetwhat/checks/check_funcs.py [hunk 5]
```diff
@@ -24,12 +24,15 @@ def check_range(state, field, field_msg, missing_msg=None):
...
+                return normalize(pattern) in str(on_text)
+            else:
+                return len(re.findall(normalize(pattern), str(on_text))) >= (index + 1)
```
Comment (hermansje) [inline sheetwhat/checks/check_funcs.py]: Does this work if the pattern contains regex characters (which is one of the use cases for `fixed=True`)?

File: sheetwhat/checks/check_funcs.py [hunk 6]
```diff
@@ -24,12 +24,15 @@ def check_range(state, field, field_msg, missing_msg=None):
...
         else:
             return re.search(pattern, str(on_text)) is not None
```
Comment (hermansje) [inline sheetwhat/checks/check_funcs.py]: We would need an implementation here as well, maybe wrapping with something like `(?:___){___}`.

File: tests/test_check_function.py (hunk 1 above)
Comment (BogdanFloris) [reply tests/test_check_function.py]: Done

File: tests/test_check_function.py (hunk 4 above)
Comment (BogdanFloris) [reply tests/test_check_function.py]: Done

**Pull request merged**